### PR TITLE
implement ping test

### PR DESCRIPTION
**Pull request by hvnsweeting**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
...
+	}
+
+	return fmt.Sprintf("Event %s at %s", s, e.When)
+}
+func checkPing(h Host) Event {
+	out, err := exec.Command("nmap", "-sP", h.IP).Output()
+	if err == nil {
+		if strings.Contains(string(out), "1 host up") {
+			return Event{time.Now(), true}
+		}
```
Comment (meomap) [inline main.go]: Codestyle 
`Event{When:time.Now(), Up:true}`

File: main.go [hunk 2]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
...
+	}
+	return Event{time.Now(), false}
```
Comment (meomap) [inline main.go]: `Event{When:time.Now()}`

Comment (hvnsweeting) [inline main.go]: this requires to know default value when reading code

File: main.go [hunk 3]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
...
 
+type Event struct {
+	When time.Time
+	Up   bool
+}
+
+func (e Event) String() string {
+	var s string
+	if e.Up {
+		s = "UP"
```
Comment (meomap) [inline main.go]: define constant here
`
constant (
    statusUp = "UP"
    statusDown= "DOWN"
)
`

File: main.go [hunk 4]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
...
+		chost <- host
+	}
+	close(ch)
+	close(chost)
+}
+
 func main() {
 	hosts := arpEntries()
-	display(hosts)
+	ch := make(chan Event)
```
Comment (meomap) [inline main.go]: `ch` & `chost`
what does those mean, should be `eventChan` & `hostChan` for clarification

File: main.go [hunk 5]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
...
+	out, err := exec.Command("nmap", "-sP", h.IP).Output()
+	if err == nil {
+		if strings.Contains(string(out), "1 host up") {
+			return Event{time.Now(), true}
+		}
+	}
+	return Event{time.Now(), false}
+}
+
+func checkAll(hosts []Host, ch chan Event, chost chan Host) {
```
Comment (meomap) [inline main.go]: Why does this function needs another channel to store `Host` while its input is also a list of `Host`

File: main.go [hunk 6]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
...
+	for _, host := range hosts {
+		ch <- checkPing(host)
+		chost <- host
```
Comment (meomap) [inline main.go]: pas fin
`Host` info should be associated with `Event`, not two separated channel with same index like this

File: main.go [hunk 7]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
 	return "", err
 }
 
+func nmapScanAll() {
```
Comment (meomap) [inline main.go:108]: return `error`, delegate error handling to main entry

File: main.go [hunk 8]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
...
+	myIP, err := myLANIP()
+	handleErr(err)
+	out, err := exec.Command("nmap", "-A", "-T4", myIP).Output()
+	handleErr(err)
+	log.Println(string(out))
+}
 func nmapEntries() {
```
Comment (meomap) [inline main.go:115]: return `error`, delegate error handling to main entry

File: main.go [hunk 9]
```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
...
+func checkAll(hosts []Host, ch chan Event, chost chan Host) {
+	for _, host := range hosts {
+		ch <- checkPing(host)
+		chost <- host
+	}
+	close(ch)
+	close(chost)
+}
+
 func main() {
```
Comment (meomap) [inline main.go:151]: error handling should be in main entry

**Pull request merged**